tests/testthat/test-round_type.R

```diff
@@ -0,0 +1,348 @@
+context("Printing tables with proper round_type")
+
+prep_exp_str <- function(colheader, txtvals, txt, totl = 28, len = 6) {
+  colheader <- substr(paste(strrep(" ", len), paste(sprintf("%-7s", colheader), collapse = " ")), 1, totl)
+  txtvals <- paste(substr(paste(txt, paste(sprintf("%-7s", txtvals), collapse = " ")), 1, totl - 1), "\n")
+
+  expstr_lns <- c(
+    colheader,
+    strrep("—", totl),
+    txtvals
+  )
+
+  exp_str <- paste(expstr_lns, collapse = "\n")
+  exp_str
+}
+
+txtvals_iec <- vals_round_type_fmt(vals = vals_round_type, round_type = "iec")
+txtvals_sas <- vals_round_type_fmt(vals = vals_round_type, round_type = "sas")
+
+# confirm that at least one value is different
+expect_true(any(txtvals_iec != txtvals_sas))
+
+test_that("round_type can be set on basic_table", {
+  skip_if_not_installed("dplyr")
+
+  tbl_sas <- tt_to_test_round_type(round_type = "sas")
+
+  expect_identical(
+    obj_round_type(tbl_sas),
+    "sas"
+  )
+
+  # rounding method can be changed without the need to rebuild the table
+  tbl_iec <- tbl_sas
+  obj_round_type(tbl_iec) <- "iec"
+
+  expect_identical(
+    obj_round_type(tbl_iec),
+    "iec"
+  )
+
+  # disallowed rounding method will give error
+  tbl_fake <- tbl_sas
+  expect_error(obj_round_type(tbl_fake) <- "fake")
+
+  # actual formatted values are as required
+  colheader <- c("ARM A", "ARM B", "ARM C")
+  names(txtvals_iec) <- colheader
+  names(txtvals_sas) <- colheader
+
+  exp_str_iec <- prep_exp_str(colheader, txtvals_iec, "Mean  ")
+  exp_str_sas <- prep_exp_str(colheader, txtvals_sas, "Mean  ")
+
+  expect_identical(
+    toString(tbl_iec),
+    exp_str_iec
+  )
+
+  expect_identical(
+    toString(tbl_sas),
+    exp_str_sas
+  )
+})
+
```

Comment:
Add tests for overriding round type in toString/export_as_txt

```
toString(..., round_type = "iec_mod")
```

Add test for matrix_form method (make sure it ends up with the right round_type)